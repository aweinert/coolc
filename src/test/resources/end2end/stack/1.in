2
4
+
e
d
x
