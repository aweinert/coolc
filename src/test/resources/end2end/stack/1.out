This is a very simple stack machine that only supports the following commands:
<int> - Push the given integer on top of the stack. Cannot be evaluated.
+     - Push an addition command on the stack. When evaluated, it takes the next two elements from the stack,
        which must be ints, adds them and pushes the result back on the stack.
s     - Push a swap command on the stack. When evaluated, this takes the next two elements from the stack,
        and pushes them back on the stack in reverse order.
e     - Evaluate the command on top of the stack.
d     - Display the current stack.
x     - Quit the program.
>>>>>6
>